libvast/test/detail/legacy_deserialize.cpp

```diff
@@ -0,0 +1,346 @@
+  CHECK_EQUAL(ui_max, ui_max2);
+  // unsigned long max
+  buf.clear();
+  const unsigned long uli_max = ULONG_MAX;
+  CHECK_EQUAL(detail::serialize(buf, uli_max), caf::none);
+  unsigned long uli_max2 = 0;
+  REQUIRE(ldes(buf, uli_max2));
+  CHECK_EQUAL(uli_max2, uli_max);
+  // unsigned long long max
+  buf.clear();
+  const unsigned long long ulli_max = ULLONG_MAX;
+  CHECK_EQUAL(detail::serialize(buf, ulli_max), caf::none);
+  unsigned long long ulli_max2 = 0;
+  REQUIRE(ldes(buf, ulli_max2));
+  CHECK_EQUAL(ulli_max, ulli_max2);
+  // char min
+  buf.clear();
+  const char c_min = CHAR_MIN;
+  CHECK_EQUAL(detail::serialize(buf, c_min), caf::none);
+  char c_min2 = CHAR_MAX;
+  REQUIRE(ldes(buf, c_min2));
+  CHECK_EQUAL(c_min, c_min2);
+  // char max
+  buf.clear();
+  const char c_max = CHAR_MAX;
+  CHECK_EQUAL(detail::serialize(buf, c_max), caf::none);
+  char c_max2 = 0;
+  REQUIRE(ldes(buf, c_max2));
+  CHECK_EQUAL(c_max, c_max2);
+  // bool
+  buf.clear();
+  const bool b = false;
+  CHECK_EQUAL(detail::serialize(buf, b), caf::none);
+  bool b2 = true;
+  REQUIRE(ldes(buf, b2));
+  CHECK_EQUAL(b, b2);
+}
+
+TEST(deserialize_bytes) {
+  const std::array bytes{std::byte{'a'}, std::byte{'c'}};
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, bytes), caf::none);
+  std::array<std::byte, 2> bytes2{};
+  REQUIRE(ldes(buf, bytes2));
+  CHECK_EQUAL(bytes, bytes2);
+}
+
+TEST(deserialize_record_type) {
+  const auto r = legacy_record_type{
+    {
+      "x",
+      legacy_record_type{
+        {"y",
+         legacy_record_type{
+           {"z", legacy_integer_type{}},
+           {"k", legacy_bool_type{}},
+         }},
+        {"m",
+         legacy_record_type{
+           {"y",
+            legacy_record_type{
+              {"a", legacy_address_type{}},
+            }},
+           {"f", legacy_real_type{}},
+         }},
+        {"b", legacy_bool_type{}},
+      },
+    },
+    {
+      "y",
+      legacy_record_type{
+        {"b", legacy_bool_type{}},
+      },
+    },
+  };
+
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, r), caf::none);
+  auto r2 = legacy_record_type{};
+  REQUIRE(ldes(buf, r2));
+  CHECK_EQUAL(r, r2);
+}
+
+TEST(deserialize_qualified_record_field) {
+  const auto field = qualified_record_field{
+    "zeek.conn",
+    record_field{
+      "conn.id",
+      legacy_address_type{},
+    },
+  };
+
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, field), caf::none);
+  auto field2 = qualified_record_field{};
+  REQUIRE(ldes(buf, field2));
+  CHECK_EQUAL(field, field2);
+}
+
+TEST(deserialize_ids) {
+  auto i = ids{};
+  i.append_bits(true, 20);
+  i.append_bits(false, 5);
+  i.append_bits(true, 1);
+  std::vector<char> buf;
+  CHECK_EQUAL(detail::serialize(buf, i), caf::none);
+  auto i2 = ids{};
+  REQUIRE(ldes(buf, i2));
+  CHECK_EQUAL(i, i2);
+}
+
+auto to_addr_view(std::string_view str) {
+  return make_data_view(unbox(to<address>(str)));
+}
```

Comment:
This has potential ODR issues, and needs to be wrapped in an anonymous namespace like so:


```suggestion
namespace {

auto to_addr_view(std::string_view str) {
  return make_data_view(unbox(to<address>(str)));
}

} // namespace
```